widget.py

```diff
@@ -100,6 +100,12 @@ def __init__(self, parent=None):
         self.sim_timer.timeout.connect(self.generate_points)
         self.sim_timer.start(25)
 
+        #QTimer to help us only keep the recent 30s data
+        self.time_range = 30000
+        self.data_filter_timer = QTimer(self)
+        self.data_filter_timer.timeout.connect(self.filter_data)
+        self.data_filter_timer.start(self.time_range)
```

Comment:
I think you might be confused on the workings of QTimer. When a QTimer is started, the passed in time is delay between timer calls in milliseconds. In this case, the timer will call `filter_data` every 30s which is why you might be seeing the delay.

A better approach for maintaining the most recent 30s of measurements might be to have this function once every 1s (1000ms) or 0.5s. We can maintain the most recent time measurement, then remove all points from the plots whose time value is less than (most recent time) - 30s.